esphome/components/uptime/text_sensor/uptime_timestamp_text_sensor.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include "esphome/core/defines.h"
+
+#ifdef USE_TIME
+
+#include "esphome/components/text_sensor/text_sensor.h"
+#include "esphome/core/component.h"
+#include "esphome/components/time/real_time_clock.h"
+
+namespace esphome {
+namespace uptime {
+
+class UptimeTimestampTextSensor : public text_sensor::TextSensor, public Component {
+ public:
+  void dump_config() override;
+  void setup() override;
+
+  float get_setup_priority() const override;
+
+  void set_time(time::RealTimeClock *time) { this->time_ = time; }
+
+ protected:
+  time::RealTimeClock *time_;
```

Comment:
```suggestion
  time::RealTimeClock *time_{};
```
Or even better, make it a constructor parameter so it's always initialised.

Reply:
if I make it a constructor parameter how would I hand over the the pointer to the rtc? 
All the components seem to allways use a setter to do stuff like that. 
Even the `uptime` `sensor` is doing it the same way.